webpack.config.ts

```diff
@@ -1,12 +1,14 @@
-'use strict';
-const path = require('path');
+import path from 'path';
+import CopyWebpackPlugin from 'copy-webpack-plugin';
+import TerserPlugin from 'terser-webpack-plugin';
+import MiniCssExtractPlugin from 'mini-css-extract-plugin';
+import webpack from 'webpack';
+
+// eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
 const SizePlugin = require('size-plugin');
-const CopyWebpackPlugin = require('copy-webpack-plugin');
-const TerserPlugin = require('terser-webpack-plugin');
-const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 
-module.exports = (env, argv) => ({
-	devtool: 'sourcemap',
+module.exports = (_env: string, argv: Record<string, boolean | number | string>): webpack.Configuration => ({
```

Comment:
Are you sure `argv` is `boolean | number | string`? I'm pretty sure it's just `string`.

Reply:
It's webpack doing something because the type for `argv.watch` is boolean. I'll double check this type tomorrow. Time for 🛏️

Reply:
`webpack` parses all arguments to normalize them, so `webpack --production` is received as `{mode: 'production'}`. Maybe even `...{watch: false, etc}`

Reply:
Yeah, I double checked this morning, they're manipulating `argv` like you said, because normally `process.argv`'s type is `string[]`. The do the same for env like I mention here, https://github.com/sindresorhus/refined-github/pull/1987#discussion_r278762904.